File: sample-java/src/main/java/net/dudmy/sample_java/BaseActivity.java
package net.dudmy.sample_java;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import net.dudmy.tutorialpop.TutorialPopHelper;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Check current page need tutorial.
        TutorialPopHelper.INSTANCE.showTutorialIfNeed(getClass().getSimpleName());
    }
}

File: sample-kotlin/src/main/java/net/dudmy/sample_kotlin/BaseActivity.kt
package net.dudmy.sample_kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import net.dudmy.tutorialpop.TutorialPopHelper

open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Check current page need tutorial.
        TutorialPopHelper.showTutorialIfNeed(javaClass.simpleName)
    }
}
